rpc/types/src/misc/distribution.rs

```diff
@@ -0,0 +1,298 @@
+fn decompress_integer_array(array: Vec<u64>) -> Vec<u64> {
+    todo!()
+}
+
+//---------------------------------------------------------------------------------------------------- Distribution
+#[doc = crate::macros::monero_definition_link!(
+    cc73fe71162d564ffda8e549b79a350bca53c454,
+    "rpc/core_rpc_server_commands_defs.h",
+    2468..=2508
+)]
+/// Used in [`crate::json::GetOutputDistributionResponse`].
+///
+/// This enum's variant depends upon the `binary` and `compress` fields.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(untagged))]
+pub enum Distribution {
+    /// Distribution data will be (de)serialized as either JSON or binary (uncompressed).
+    Uncompressed {
+        start_height: u64,
+        base: u64,
+        /// TODO: this is a binary JSON string if `binary == true`.
+        ///
+        /// Considering both the `binary` field and `/get_output_distribution.bin`
+        /// endpoint are undocumented in the first place, Cuprate could just drop support for this.
+        distribution: Vec<u64>,
+        amount: u64,
+        binary: bool,
+        #[cfg_attr(feature = "serde", serde(serialize_with = "serde_false"))]
+        /// Will always be serialized as `false`.
+        compress: bool,
+    },
+    /// Distribution data will be (de)serialized as compressed binary.
+    CompressedBinary {
+        start_height: u64,
+        base: u64,
+        compressed_data: String,
+        amount: u64,
+        #[cfg_attr(feature = "serde", serde(serialize_with = "serde_true"))]
+        /// Will always be serialized as `true`.
+        binary: bool,
+        #[cfg_attr(feature = "serde", serde(serialize_with = "serde_true"))]
+        /// Will always be serialized as `true`.
+        compress: bool,
+    },
+}
+
+impl Default for Distribution {
+    fn default() -> Self {
+        Self::Uncompressed {
+            start_height: u64::default(),
+            base: u64::default(),
+            distribution: Vec::<u64>::default(),
+            amount: u64::default(),
+            binary: false,
+            compress: false,
+        }
+    }
+}
+
+//---------------------------------------------------------------------------------------------------- Epee
+#[cfg(feature = "epee")]
+/// [`EpeeObjectBuilder`] for [`Distribution`].
+///
+/// Not for public usage.
+#[allow(dead_code, missing_docs)]
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct __DistributionEpeeBuilder {
+    pub start_height: Option<u64>,
+    pub base: Option<u64>,
+    pub distribution: Option<Vec<u64>>,
+    pub amount: Option<u64>,
+    pub compressed_data: Option<String>,
+    pub binary: Option<bool>,
+    pub compress: Option<bool>,
+}
+
+#[cfg(feature = "epee")]
+impl EpeeObjectBuilder<Distribution> for __DistributionEpeeBuilder {
```

Comment:
Sidenote: are the custom epee implementations correct in this PR?

Is `if $field.should_write() { write() }` required? Should it be `write_field`? or just `write`?

Reply:
only `write_field` needs to be called, `should_write` is checked in that function.

Everything else looks ok